File: eppo/src/main/java/cloud/eppo/android/dto/PrecomputedFlag.java
package cloud.eppo.android.dto;

import androidx.annotation.Nullable;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.Map;

/**
 * Represents a precomputed flag assignment from the edge endpoint. All string fields except
 * variationType are Base64 encoded.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class PrecomputedFlag {

  @Nullable private final String allocationKey;
  @Nullable private final String variationKey;
  private final String variationType;
  private final String variationValue;
  @Nullable private final Map<String, String> extraLogging;
  private final boolean doLog;

  @JsonCreator
  public PrecomputedFlag(
      @JsonProperty("allocationKey") @Nullable String allocationKey,
      @JsonProperty("variationKey") @Nullable String variationKey,
      @JsonProperty("variationType") String variationType,
      @JsonProperty("variationValue") String variationValue,
      @JsonProperty("extraLogging") @Nullable Map<String, String> extraLogging,
      @JsonProperty("doLog") boolean doLog) {
    this.allocationKey = allocationKey;
    this.variationKey = variationKey;
    this.variationType = variationType;
    this.variationValue = variationValue;
    this.extraLogging = extraLogging;
    this.doLog = doLog;
  }

  /** Returns the Base64-encoded allocation key, or null if not assigned. */
  @Nullable
  public String getAllocationKey() {
    return allocationKey;
  }

  /** Returns the Base64-encoded variation key, or null if not assigned. */
  @Nullable
  public String getVariationKey() {
    return variationKey;
  }

  /** Returns the variation type (STRING, BOOLEAN, INTEGER, NUMERIC, JSON). */
  public String getVariationType() {
    return variationType;
  }

  /** Returns the Base64-encoded variation value. */
  public String getVariationValue() {
    return variationValue;
  }

  /** Returns the Base64-encoded extra logging map, or null if not present. */
  @Nullable
  public Map<String, String> getExtraLogging() {
    return extraLogging;
  }

  /** Returns whether this assignment should be logged. */
  public boolean isDoLog() {
    return doLog;
  }
}

File: eppo-kotlin/src/main/kotlin/cloud/eppo/kotlin/internal/network/dto/PrecomputedFlag.kt
package cloud.eppo.kotlin.internal.network.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Precomputed flag data from the server.
 *
 * All string fields (except variationType and doLog) are Base64 encoded.
 *
 * @property flagKey MD5 hash of the flag key (optional, may be null)
 * @property allocationKey Base64 encoded allocation key
 * @property variationKey Base64 encoded variation key
 * @property variationType Type of the variation value (BOOLEAN, INTEGER, NUMERIC, STRING, JSON)
 * @property variationValue Base64 encoded variation value
 * @property extraLogging Base64 encoded extra logging metadata (keys and values)
 * @property doLog Whether to log this assignment
 */
@Serializable
internal data class PrecomputedFlag(
    @SerialName("flagKey") val flagKey: String? = null,
    @SerialName("allocationKey") val allocationKey: String? = null,
    @SerialName("variationKey") val variationKey: String? = null,
    @SerialName("variationType") val variationType: String,
    @SerialName("variationValue") val variationValue: String,
    @SerialName("extraLogging") val extraLogging: Map<String, String>? = null,
    @SerialName("doLog") val doLog: Boolean
)
